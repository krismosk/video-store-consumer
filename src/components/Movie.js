import React, { Component } from 'react';

const Movie = (props) => {
  
  return(
    <section>
      <img src={props.image} alt={`${props.title}`}/>
      <p> {props.title} </p>
      <p> {props.overview} </p>
      <p> {props.releaseDate} </p>
      <p> {props.inventory} </p>
      
      

      <button 
        type='button'
        onClick={() => {props.findMovie(props.id)}}
      >
      Select</button>
    </section>
  )
}

export default Movie;