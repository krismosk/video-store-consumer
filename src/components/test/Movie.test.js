import React from 'react';
import { render, cleanup, getByText } from '@testing-library/react';
import Movie from '../Movie';

const movie = <Movie 
  id={ 1337 }
  title={'test title'}
  image={'test image'}
  overview={'test overview'}
  releaseDate={'test date'}
  inventory={1}
  findMovie={() => { }}
/>;

describe('Movie', () => {
  test('it can be rendered and matches the snapshot', () => {
    const { asFragment } = render( movie );
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });
  test("it will render the proper data for the movie", () => {
    const container = render(movie)
    expect(container.getByText(/test title/)).toBeDefined();
    expect(container.getByText(/test overview/)).toBeDefined();
    expect(container.getByText(/test date/)).toBeDefined();
  });
  test('the findMovie function is called when the select button is clicked on', () => {
    const findMovie = jest.fn();
    const container = render(<Movie 
      id={ 1337 }
      title={'test title'}
      image={'test image'}
      overview={'test overview'}
      releaseDate={'test date'}
      inventory={1}
      findMovie={findMovie}
    />);

    const selectButton = container.getByText(/Select/);
    selectButton.click();
    expect(findMovie).toHaveBeenCalled();
  })
});