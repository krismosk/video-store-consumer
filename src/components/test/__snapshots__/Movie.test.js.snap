// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie it can be rendered and matches the snapshot 1`] = `
<DocumentFragment>
  <section>
    <img
      alt="test title"
      src="test image"
    />
    <p>
       test title 
    </p>
    <p>
       test overview 
    </p>
    <p>
       test date 
    </p>
    <p>
       1 
    </p>
    <button
      type="button"
    >
      Select
    </button>
  </section>
</DocumentFragment>
`;
